### PR TITLE
Grand renaming branch

### PR DESCRIPTION
**Pull request by Index0011**
Rebrand phala-blockchain by renaming mining contents.

### CODE REVIEW

File: crates/phala-types/src/lib.rs [hunk 1]
```diff
@@ -634,9 +634,9 @@ pub enum WorkerStateEnum<BlockNumber> {
     Empty,
     Free,
     Gatekeeper,
-    MiningPending,
-    Mining(BlockNumber),
-    MiningStopping,
+    WorkingPending,
```
Comment (jasl) [inline crates/phala-types/src/lib.rs:637]: just `Pending` is enough?

File: crates/phala-types/src/lib.rs [hunk 2]
```diff
@@ -634,9 +634,9 @@ pub enum WorkerStateEnum<BlockNumber> {
...
+    Computing(BlockNumber),
+    WorkingStopping,
```
Comment (jasl) [inline crates/phala-types/src/lib.rs:639]: `Stopping`

File: crates/phala-types/src/lib.rs [hunk 3]
```diff
@@ -258,26 +258,26 @@ pub mod messaging {
...
         ///  The init bench score caculated by pallet.
         BenchScore(u32),
-        /// pallet-mining --> worker
-        ///  When a miner start to mine, push this message to the worker to start the benchmark task.
-        ///   session_id: Generated by pallet. Each mining session should have a unique session_id.
-        MiningStart {
+        /// pallet-computing --> worker
+        ///  When a worker start to computing, push this message to the worker to start the benchmark task.
+        ///   session_id: Generated by pallet. Each computing session should have a unique session_id.
+        WorkingStarted {
```
Comment (jasl) [inline crates/phala-types/src/lib.rs:264]: `Started`

File: crates/phala-types/src/lib.rs [hunk 4]
```diff
@@ -258,26 +258,26 @@ pub mod messaging {
...
         },
-        /// pallet-mining --> worker
-        ///  When a miner entered CoolingDown state, push this message to the worker, so that it can stop the
+        /// pallet-computing --> worker
+        ///  When a worker entered CoolingDown state, push this message to the worker, so that it can stop the
         ///  benchmark task.
-        MiningStop,
-        /// pallet-mining --> worker
-        ///  When a miner entered Unresponsive state, push this message to the worker to suppress the subsequent
+        WorkingStopped,
```
Comment (jasl) [inline crates/phala-types/src/lib.rs:272]: `Stopped`

File: crates/phala-types/src/lib.rs [hunk 5]
```diff
@@ -258,26 +258,26 @@ pub mod messaging {
...
         ///  heartbeat responses.
-        MiningEnterUnresponsive,
-        /// pallet-mining --> worker
-        ///  When a miner recovered to MiningIdle state from Unresponsive, push this message to the worker to
+        WorkerEnterUnresponsive,
+        /// pallet-computing --> worker
+        ///  When a worker recovered to WorkerIdle state from Unresponsive, push this message to the worker to
         ///  resume the subsequent heartbeat responses.
-        MiningExitUnresponsive,
+        WorkerExitUnresponsive,
```
Comment (jasl) [inline crates/phala-types/src/lib.rs:280]: `ExitUnresponsive`

**Comment by jasl** [comment]
Overall LGTM, I propose simplify some redundents

**Comment by kvinwang** [comment]
Is the branch vault3 going to be merged before the next pRuntime release (for DogFood)?
If not, better rebasing this PR onto the master because it would break the checkpoint of pRuntime.

File: crates/phactory/src/prpc_service.rs [hunk 6]
```diff
@@ -938,7 +938,7 @@ impl<Platform: pal::Platform + Serialize + DeserializeOwned> PhactoryApi for Rpc
             .try_into()
             .map_err(|_| from_display("Bad public key"))?;
         let state = gk
-            .mining_economics
```
Comment (doyleguo) [inline crates/phactory/src/prpc_service.rs:941]: Maybe Rewards_Economics is better?

File: standalone/replay/src/replay_gk.rs [hunk 7]
```diff
@@ -44,15 +44,15 @@ pub struct ReplayFactory {
     #[serde(skip)]
     #[serde(default)]
     recv_mq: MessageDispatcher,
-    gk: gk::MiningEconomics<ReplayMsgChannel>,
+    gk: gk::WorkingEconomics<ReplayMsgChannel>,
```
Comment (doyleguo) [inline standalone/replay/src/replay_gk.rs:47]: 同上

File: crates/phactory/src/prpc_service.rs (hunk 6 above)
Comment (h4x3rotab) [reply crates/phactory/src/prpc_service.rs:941]: compute_economic?

File: standalone/runtime/src/lib.rs [hunk 8]
```diff
@@ -610,14 +610,14 @@ parameter_types! {
...
+	// worker configs
 	pub const MultiPhaseUnsignedPriority: TransactionPriority = StakingUnsignedPriority::get() - 1u64;
-	pub MinerMaxWeight: Weight = RuntimeBlockWeights::get()
+	pub WorkerMaxWeight: Weight = RuntimeBlockWeights::get()
 		.get(DispatchClass::Normal)
 		.max_extrinsic.expect("Normal extrinsics have a weight limit configured; qed")
 		.saturating_sub(BlockExecutionWeight::get());
 	// Solution can occupy 90% of normal block size
-	pub MinerMaxLength: u32 = Perbill::from_rational(9u32, 10) *
+	pub WorkerMaxLength: u32 = Perbill::from_rational(9u32, 10) *
```
Comment (h4x3rotab) [inline standalone/runtime/src/lib.rs:620]: These parameters belong to `frame_election_privider`. We don't need to change.

File: standalone/runtime/src/msg_routing.rs [hunk 9]
```diff
@@ -32,8 +32,8 @@ impl pallet_mq::QueueNotifyConfig for MessageRouteConfig {
         route_handlers! {
             PhalaRegistry::on_message_received,
             PhalaRegistry::on_gk_message_received,
-            PhalaMining::on_gk_message_received,
-            PhalaMining::on_mining_message_received,
+            PhalaComputation::on_gk_message_received,
+            PhalaComputation::working_message_received,
```
Comment (h4x3rotab) [inline standalone/runtime/src/msg_routing.rs:36]:
```suggestion
            PhalaComputation::on_working_message_received,
```

Missed "on"?

File: pallets/phala/src/compute/computation.rs [hunk 10]
```diff
@@ -40,25 +40,25 @@ pub mod pallet {
 	use fixed_sqrt::FixedSqrt;
 
 	const DEFAULT_EXPECTED_HEARTBEAT_COUNT: u32 = 20;
-	const MINING_PALLETID: PalletId = PalletId(*b"phala/pp");
+	const COMPUTING_PALLETID: PalletId = PalletId(*b"phala/pp");
 
 	#[derive(Encode, Decode, TypeInfo, Clone, PartialEq, Eq, RuntimeDebug)]
-	pub enum MinerState {
+	pub enum WorkerState {
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/computation.rs:46]: Maybe call it "SessionState", short for "ComputingSessionSatate". Not quite sure. Just leave a comment for now

File: pallets/phala/src/compute/computation.rs [hunk 11]
```diff
@@ -123,11 +123,11 @@ pub mod pallet {
 		}
 	}
 
-	/// The state of a miner
+	/// The state of a worker
 	#[derive(Encode, Decode, TypeInfo, Clone, PartialEq, Eq, RuntimeDebug)]
-	pub struct MinerInfo {
-		/// The current state of the miner
-		pub state: MinerState,
+	pub struct WorkerInfo {
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/computation.rs:128]: Better "SessionInfo" or "ComputeInfo". In this file, "worker" usually refers to the physical worker, and "miner" refers to the mining session.

File: pallets/phala/src/compute/computation.rs [hunk 12]
```diff
@@ -254,21 +255,21 @@ pub mod pallet {
...
-	/// The next id to assign to a mining session
+	/// The next id to assign to a computing session
 	#[pallet::storage]
 	pub type NextSessionId<T> = StorageValue<_, u32, ValueQuery>;
 
-	/// The block number when the mining starts. Used to calculate halving.
+	/// The block number when the computing starts. Used to calculate halving.
 	#[pallet::storage]
-	pub type MiningStartBlock<T: Config> = StorageValue<_, T::BlockNumber>;
+	pub type WorkingStartBlock<T: Config> = StorageValue<_, T::BlockNumber>;
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/computation.rs:264]:
```suggestion
	pub type ComputeStartBlock<T: Config> = StorageValue<_, T::BlockNumber>;
```

Here "Mining" refer to the mining mechanism. I think it's better to replace it by "compute".

File: pallets/phala/src/compute/computation.rs [hunk 13]
```diff
@@ -254,21 +255,21 @@ pub mod pallet {
...
 
 	/// The interval of halving (75% decay) in block number.
 	#[pallet::storage]
-	pub type MiningHalvingInterval<T: Config> = StorageValue<_, T::BlockNumber>;
+	pub type WorkingHalvingInterval<T: Config> = StorageValue<_, T::BlockNumber>;
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/computation.rs:268]: ditto

File: pallets/phala/src/compute/computation.rs [hunk 14]
```diff
@@ -280,71 +281,71 @@ pub mod pallet {
...
-		/// Miner & worker are bound.
+		/// Worker & worker are bound.
 		///
 		/// Affected states:
-		/// - [`MinerBindings`] for the miner account is pointed to the worker
-		/// - [`WorkerBindings`] for the worker is pointed to the miner account
-		/// - the miner info at [`Miners`] is updated with `Ready` state
-		MinerBound {
-			miner: T::AccountId,
+		/// - [`SessionBindings`] for the worker account is pointed to the worker
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/computation.rs:311]: “the miner account” is the virtual account generated for the mining session. So we'd better say "for the session account"

File: pallets/phala/src/compute/computation.rs [hunk 15]
```diff
@@ -280,71 +281,71 @@ pub mod pallet {
...
+		/// Worker settled successfully.
 		///
-		/// It results in the v in [`Miners`] being updated. It also indicates the downstream
-		/// stake pool has received the mining reward (payout), and the treasury has received the
+		/// It results in the v in [`Workers`] being updated. It also indicates the downstream
+		/// stake pool has received the computing reward (payout), and the treasury has received the
 		/// tax.
-		MinerSettled {
-			miner: T::AccountId,
+		WorkerSettled {
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/computation.rs:342]: So this should be "SessionSettled"

File: pallets/phala/src/compute/computation.rs [hunk 16]
```diff
@@ -380,33 +384,33 @@ pub mod pallet {
 		WorkerNotRegistered,
 		/// Deprecated
 		_GatekeeperNotRegistered,
-		/// Not permitted because the miner is already bound with another worker.
-		DuplicateBoundMiner,
+		/// Not permitted because the worker is already bound with another worker.
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/computation.rs:387]: > the worker is already bound with another worker.

the session account is already bound with another worker

File: pallets/phala/src/compute/computation.rs [hunk 17]
```diff
@@ -572,36 +576,36 @@ pub mod pallet {
...
+						// assume the worker is always there and the worker state is complete. So
 						// far it sounds safe to just discard this message, but not interrupt the
 						// entire message queue.
 						//
 						// So we call `ensure_worker_bound` here, and return an error if the worker
 						// is not bound. However if the worker is indeed bound, the rest of the
-						// code assumes the Miners, Workers, and worker score must exist.
-						let miner = Self::ensure_worker_bound(&worker)?;
-						let mut miner_info = Self::miners(&miner).expect("Bound miner; qed.");
+						// code assumes the Workers, Workers, and worker score must exist.
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/computation.rs:585]: > Workers, Workers, and worker score must exist.

Sessions, Workers, and worker score

File: pallets/phala/src/compute/computation.rs [hunk 18]
```diff
@@ -572,36 +576,36 @@ pub mod pallet {
...
+						let session = Self::ensure_worker_bound(&worker)?;
+						let mut worker_info = Self::sessions(&session).expect("Bound worker; qed.");
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/computation.rs:587]:
```suggestion
						let mut session_info = Self::sessions(&session).expect("Bound worker; qed.");
```

File: pallets/phala/src/compute/stakepoolv2.rs [hunk 19]
```diff
@@ -64,16 +64,16 @@ pub mod pallet {
 		#[pallet::constant]
 		type GracePeriod: Get<u64>;
 
-		/// If mining is enabled by default.
+		/// If computing is enabled by default.
 		#[pallet::constant]
-		type MiningEnabledByDefault: Get<bool>;
+		type WorkingEnabledByDefault: Get<bool>;
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/stakepoolv2.rs:69]:
```suggestion
		type ComputingEnabledByDefault: Get<bool>;
```

File: pallets/phala/src/compute/stakepoolv2.rs [hunk 20]
```diff
@@ -64,16 +64,16 @@ pub mod pallet {
...
 
 		/// The max allowed workers in a pool
 		#[pallet::constant]
 		type MaxPoolWorkers: Get<u32>;
 
-		/// The origin that can turn on or off mining
-		type MiningSwitchOrigin: EnsureOrigin<Self::Origin>;
+		/// The origin that can turn on or off computing
+		type WorkingSwitchOrigin: EnsureOrigin<Self::Origin>;
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/stakepoolv2.rs:76]:
```suggestion
		type ComputingSwitchOrigin: EnsureOrigin<Self::Origin>;
```

File: pallets/phala/src/compute/stakepoolv2.rs [hunk 21]
```diff
@@ -729,8 +728,8 @@ pub mod pallet {
 			let grace_period = T::GracePeriod::get();
 			let mut releasing_stake = Zero::zero();
 			for worker in pool.cd_workers.iter() {
-				let miner: T::AccountId = pool_sub_account(pid, worker);
-				let stakes: BalanceOf<T> = mining::pallet::Stakes::<T>::get(&miner)
+				let worker: T::AccountId = pool_sub_account(pid, worker);
```
Comment (h4x3rotab) [inline pallets/phala/src/compute/stakepoolv2.rs:731]:
```suggestion
				let session: T::AccountId = pool_sub_account(pid, worker);
```

**Comment by h4x3rotab** [comment]
> Is the branch vault3 going to be merged before the next pRuntime release (for DogFood)? If not, It's better to rebase this PR onto the master because it would break the checkpoint of pRuntime.

What does break the checkpoint? Field name in serialization?

**Comment by kvinwang** [comment]
> Field name in serialization?

Yep

**Comment by kvinwang** [comment]
> Is the branch vault3 going to be merged before the next pRuntime release (for DogFood)? If not, It's better to rebase this PR onto the master because it would break the checkpoint of pRuntime.

Or we can rename the fields in gk.rs first on the master if it's hard to rebase this PR.

**Pull request merged**